packages/driver/src/cypress/log.ts

```diff
@@ -299,50 +302,63 @@ export class Log {
   }
 
   set (key, val?) {
+    let obj = key
+
     if (_.isString(key)) {
-      this.obj = {}
-      this.obj[key] = val
-    } else {
-      this.obj = key
+      obj = {}
+      obj[key] = val
     }
 
-    if ('url' in this.obj) {
+    const isHiddenLog = this.get('hidden') || obj.hidden
+
+    if ('url' in obj) {
       // always stringify the url property
-      this.obj.url = (this.obj.url != null ? this.obj.url : '').toString()
+      obj.url = (obj.url != null ? obj.url : '').toString()
     }
 
     // convert onConsole to consoleProps
     // for backwards compatibility
-    if (this.obj.onConsole) {
-      this.obj.consoleProps = this.obj.onConsole
+    if (obj.onConsole) {
+      obj.consoleProps = obj.onConsole
+      delete obj.onConsole
+    }
+
+    if (obj.message && this.config('protocolEnabled') && isHiddenLog) {
```

Comment:
Suggest adding a comment here explaining why we are truncating.

Reply:
[`ccbeea0` (#28531)](https://github.com/cypress-io/cypress/pull/28531/commits/ccbeea0bfaf24861ebb0d38c53ee6cba94bbddd2)